### PR TITLE
Faster greedy decoding and support for 3D grayscale

### PR DESCRIPTION
**Pull request by samayala22**
- MUCH faster greedy decoding function: 20-1000x (+) speedup, which in turn, speeds up the whole ocr process by 1.1-2x. The bigger the text the bigger the speedup. The difference is noticeable when scanning multiple images. One of my scans used to take ~2700s with 900s greedy decoding time. Now, the decoding part takes 1.5s (Added some comments to explain the vectorization process). On a side note, I'm not sure why there is a `for` loop for the elements of the `length` tensor (arg in `decode_greedy`), since in my testing, the tensor always had 1 element.
- Support for 3D grayscale arrays (MxNx1) .

### CODE REVIEW

**Comment by rkcosmos** [comment]
sounds very nice, i will review the code and get back soon.

**Comment by samayala22** [comment]
@rkcosmos Any updates on the PR ?

**Comment by rkcosmos** [comment]
Sorry I just have a chance to test it. I'm sure vectorization is better old for loop but I still don't see 1.1-2x speedup (whole process). Can you share 1 image with significant speedup?

**Comment by samayala22** [comment]
@rkcosmos Sure, here are the details of the speedup:

I used cProfile to profile a little python script that reads an image (screenshot from Wikipedia) and performs the OCR 20 times in a row. [Here](https://drive.google.com/drive/folders/1FSwBXujmcb0lrRH6nLEY_rwu74gGq_3f?usp=sharing) is the link to the google drive folder where you can see the image I used and the 3 prof files (then I used [snakeviz ](https://jiffyclub.github.io/snakeviz/) to visualize the results)

1. _easy_original.prof_ : original version of EasyOCR before the PR:
    - Total: 108s
    - Decode function: 41.79s (700 calls) / 0.0597s per call
2. _easy_fast.prof_: changed `preds_index.data` to `preds_index.data.cpu().detach().numpy()`
    - Total: 67.8s (1.59x faster)
    - Decode function: 0.22s (700 calls) / 3.16e-4s per call (189x faster)
3. _easy_fastest.prof_: + vectorized function
    - Total: 67.6s (1.6x faster than original)
    - Decode function: 0.054s (700 calls) / 7.66e-05s per call (780x faster than original)

As you can see, the main speedup was done by converting the gpu tensor data to a cpu numpy array. With this image, the vectorization makes the function 4x faster and the total speedup will be greater on longer texts.
